manifests/server/read_database_ini.pp

```diff
@@ -42,12 +45,16 @@
       mode   => '0600';
     }
 
+    $ini_setting_require = str2bool($database_validate) ? {
+      false => undef,
+      default  => Class['puppetdb::server::validate_read_db'],
+    }
     # Set the defaults
     Ini_setting {
       path    => "${confdir}/read_database.ini",
       ensure  => present,
       section => 'read-database',
-      require => Class['puppetdb::server::validate_db'],
```

Comment:
From the top of my head I think the change to validate_read_db is correct IMHO. If it works, I would keep it and call the old way a bug.